core/src/optimistic_confirmation_verifier.rs

```diff
@@ -153,16 +167,30 @@ mod test {
         let bank_hash = Hash::default();
         let mut optimistic_confirmation_verifier =
             OptimisticConfirmationVerifier::new(snapshot_start_slot);
-        optimistic_confirmation_verifier
-            .add_new_optimistic_confirmed_slots(vec![(snapshot_start_slot - 1, bank_hash)]);
-        optimistic_confirmation_verifier
-            .add_new_optimistic_confirmed_slots(vec![(snapshot_start_slot, bank_hash)]);
-        optimistic_confirmation_verifier
-            .add_new_optimistic_confirmed_slots(vec![(snapshot_start_slot + 1, bank_hash)]);
-        assert_eq!(optimistic_confirmation_verifier.unchecked_slots.len(), 1);
-        assert!(optimistic_confirmation_verifier
-            .unchecked_slots
-            .contains(&(snapshot_start_slot + 1, bank_hash)));
+        let blockstore_path = get_tmp_ledger_path!();
+        {
+            let blockstore = Blockstore::open(&blockstore_path).unwrap();
```

Comment:
There is a macro that you can use that will automatically clean itself up; no need for the extra inner scope or to call `Blockstore::destroy()` at the end.
https://github.com/solana-labs/solana/blob/c248fb3f51f0431801eb1cba8a8a1c11cd4bada7/ledger/src/blockstore.rs#L5805-L5806

Reply:
updated